src/js/angular/guides/guides.service.js

```diff
@@ -236,12 +236,28 @@ function GuidesService(
         if (angular.isDefined(startStepId)) {
             ShepherdService.resumeGuide(guide.guideId, stepsDescriptions, startStepId);
         } else {
-            ShepherdService.startGuide(guide.guideId, stepsDescriptions, startStepId);
+            ShepherdService.startGuide(guide.guideId, stepsDescriptions, startStepId, isAutoStarted);
         }
 
         $rootScope.$broadcast('guideStarted');
     };
 
+    this.autoStartGuide = (autoStartGuideId) => {
```

Comment:
Add normalization or check for this argument because right now you rely on the guideId and the request parameter to be of the same type which is string, but the guideId can be also a number and then the logic below might break.

Reply:
Added check and documentation

Reply:
Added check and conversion to string if `autoStartGuideId` is number.

Also added explicit conversion of `guide.guideId` to string as it is used as a further in the services.